### DISCUSSION TITLE
how to php mysql Null value check .$txQuery = $txQuery->where(function ($query) use ($label, $portfolioTxTableName) {     $query->whereNull($portfolioTxTableName . '.market'); });

### DISCUSSION THREAD

**Discussion by sriSTA00** [community/community#77841] [Programming Help] [answered]
### Body

.$txQuery = $txQuery->where(function ($query) use ($label, $portfolioTxTableName) {
    $query->whereNull($portfolioTxTableName . '.market');
});

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dakshithadissanayaka** [comment]
The provided code snippet utilizes PHP's Eloquent ORM to check for null values in a specific column named 'market' within a table named 'portfolioTxTableName'. Here's a breakdown of the code:

```php
$txQuery = $txQuery->where(function ($query) use ($label, $portfolioTxTableName) {
    $query->whereNull($portfolioTxTableName . '.market');
});
```

1. **`$txQuery = $txQuery->where(...);`**: This line initiates a `where()` clause on the existing query object `$txQuery`.

2. **`function ($query) use ($label, $portfolioTxTableName)`**: This anonymous function defines the scope of the `where()` clause, allowing access to the variables `$label` and `$portfolioTxTableName` within the function scope.

3. **`$query->whereNull($portfolioTxTableName . '.market');`**: This line within the anonymous function checks for null values in the column `market` of the table `$portfolioTxTableName`. The `whereNull()` method returns `true` if the value of the specified column is `null` and `false` otherwise.

In summary, the provided code snippet filters the `$txQuery` to include only records where the `market` column in the `portfolioTxTableName` table is `null`.

**Comment by kavindyasinthasilva** [comment, marked as answer]
If you want to check for the opposite (i.e., where the column is not NULL), you can use the whereNotNull method:

php

```
$txQuery = $txQuery->where(function ($query) use ($label, $portfolioTxTableName) {
    $query->whereNotNull($portfolioTxTableName . '.market');
});
```
This will generate a query to retrieve records where the 'market' column is not NULL. The generated SQL query will look like:

sql

`WHERE table_name.market IS NOT NULL`
Adjust the code according to your specific requirements and table structure.
